src/io/flutter/inspector/DiagnosticsNode.java

```diff
@@ -612,29 +622,73 @@ public InspectorInstanceRef getDartDiagnosticRef() {
-            for (Location pl : parameterLocations) {
-              final String name = pl.getName();
-              if (name != null) {
-                names.put(name, pl);
-              }
+          final Map<String, Location> names = new HashMap<>();
+          for (Location pl : parameterLocations) {
+            final String name = pl.getName();
+            if (name != null) {
+              names.put(name, pl);
             }
-            for (DiagnosticsNode node : nodes) {
-              node.setParent(this);
-              final String name = node.getName();
-              if (name != null) {
-                final Location parameterLocation = names.get(name);
-                if (parameterLocation != null) {
-                  node.setCreationLocation(parameterLocation);
-                }
+          }
+          for (DiagnosticsNode node : nodes) {
+            node.setParent(this);
+            final String name = node.getName();
+            if (name != null) {
+              final Location parameterLocation = names.get(name);
+              if (parameterLocation != null) {
+                node.setCreationLocation(parameterLocation);
               }
             }
+          }
         }
       }
       return nodes;
     });
   }
 
+  @NotNull
+  public CompletableFuture<String> getPropertyDoc() {
+    if (propertyDocFuture == null) {
+      propertyDocFuture = createPropertyDocFurure();
+    }
+    return propertyDocFuture;
+  }
+
+  private CompletableFuture<String> createPropertyDocFurure() {
+    final DiagnosticsNode parent = getParent();
+    if (parent != null) {
+      final InspectorService inspectorService = parent.getInspectorService();
+      final CompletableFuture<DartVmServiceValue> valueFuture =
+        inspectorService.toDartVmServiceValueForSourceLocation(parent.getValueRef());
+      return valueFuture.thenComposeAsync((DartVmServiceValue vmValue) -> {
+        final InstanceRef instanceRef = vmValue.getInstanceRef();
+        return inspectorService.getPropertyLocation(instanceRef, getName()).thenApplyAsync((XSourcePosition sourcePosition) -> {
+          if (sourcePosition != null) {
+            final VirtualFile file = sourcePosition.getFile();
+            final int offset = sourcePosition.getOffset();
+
+            final Project project = getProject(file);
+            if (project != null) {
+              final List<HoverInformation> hovers =
+                DartAnalysisServerService.getInstance(project).analysis_getHover(file, offset);
+              if (!hovers.isEmpty()) {
+                return hovers.get(0).getDartdoc();
+              }
+            }
+          }
+          return "Unable to find property source";
+        });
+      });
+    }
+
+    return CompletableFuture.completedFuture("Unable to find property source");
+  }
+
+  @Nullable
+  private Project getProject(@NotNull VirtualFile file) {
+    final FlutterDebugProcess process = inspectorService.getDebugProcess();
```

Comment:
nit: the debugProcess seems like a surprising place to be getting the app from. Perhaps have inspectorService expose
getApp() directly?

Reply:
Sounds good.  Updated.